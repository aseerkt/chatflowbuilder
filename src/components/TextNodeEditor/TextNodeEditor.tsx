import useStore from '@/stores'
import { IconArrowLeft } from '@tabler/icons-react'
import { Button } from '../ui/button'
import { Label } from '../ui/label'
import { Textarea } from '../ui/textarea'

type TextNodeEditorProps = {
  nodeId: string
}

export default function TextNodeEditor({ nodeId }: TextNodeEditorProps) {
  const node = useStore((state) =>
    state.nodes.find((node) => node.id === nodeId),
  )
  const updateNode = useStore((state) => state.updateNode)
  const deselectNode = useStore((state) => state.deselectNode)

  if (!node) return null

  const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    updateNode({
      ...node,
      data: {
        ...node.data,
        message: e.target.value,
      },
    })
  }

  return (
    <div>
      <header className='flex items-center gap-x-4 border-b px-4 py-2'>
        <Button
          variant='link'
          className='rounded-full p-0'
          aria-label='Back to node panel'
          onClick={deselectNode}
        >
          <IconArrowLeft />
        </Button>
        <span className='font-semibold'>Message</span>
      </header>
      <div className='flex flex-col gap-y-2 p-4'>
        <Label htmlFor='message'>Text</Label>
        <Textarea
          id='message'
          onChange={onChange}
          value={node.data.message || ''}
          rows={4}
        />
      </div>
    </div>
  )
}
