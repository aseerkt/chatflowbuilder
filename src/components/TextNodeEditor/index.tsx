export { default as TextNodeEditor } from './TextNodeEditor'
