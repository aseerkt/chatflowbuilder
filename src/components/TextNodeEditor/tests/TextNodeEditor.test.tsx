import useStore from '@/stores'
import { fireEvent, render } from '@testing-library/react'
import { describe, it } from 'vitest'
import TextNodeEditor from '../TextNodeEditor'

describe('<TextNodeEditor /> tests', () => {
  const nodeId = 'node_1'
  const initialMessage = 'message 1'
  beforeEach(() => {
    useStore.setState({
      nodes: [
        {
          id: nodeId,
          type: 'textNode',
          data: { message: initialMessage },
          position: { x: 100, y: 100 },
        },
      ],
      nodeId: nodeId,
      nodeType: 'textNode',
    })
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<TextNodeEditor nodeId={nodeId} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should update node message', () => {
    const { getByText } = render(<TextNodeEditor nodeId={nodeId} />)
    const textareaElement = getByText(initialMessage)
    fireEvent.input(textareaElement, { target: { value: 'message 2' } })
    expect(textareaElement).toHaveValue('message 2')
  })
})
