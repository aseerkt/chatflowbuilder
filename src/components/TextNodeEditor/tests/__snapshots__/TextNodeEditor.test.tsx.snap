// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TextNodeEditor /> tests > should match snapshot 1`] = `
<body>
  <div>
    <div>
      <header
        class="flex items-center gap-x-4 border-b px-4 py-2"
      >
        <button
          aria-label="Back to node panel"
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 rounded-full p-0"
        >
          <svg
            class="tabler-icon tabler-icon-arrow-left "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12l14 0"
            />
            <path
              d="M5 12l6 6"
            />
            <path
              d="M5 12l6 -6"
            />
          </svg>
        </button>
        <span
          class="font-semibold"
        >
          Message
        </span>
      </header>
      <div
        class="flex flex-col gap-y-2 p-4"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for="message"
        >
          Text
        </label>
        <textarea
          class="flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="message"
          rows="4"
        >
          message 1
        </textarea>
      </div>
    </div>
  </div>
</body>
`;
